### PR TITLE
Simpler evaluations part 2.2

### PR DESCRIPTION
**Pull request by csansoon**
[Part 2.2](https://github.com/latitude-dev/latitude-llm/issues/420)

Mainly, two changes:

- Update the EvaluationDto type and EvaluationRepository to fetch data from all new polymorphic relations (`metadata` and `resultConfiguration`).
- Adapt current create function to include the result configuration in both old `evaluation.metadata.configuration` and new `evaluation.resultConfiguration`. The old version will be set as legacy in the future.

### CODE REVIEW

File: packages/core/src/repositories/evaluationsRepository.ts [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
+    }),
+    ['id', 'createdAt', 'updatedAt'],
+  ),
+  resultConfiguration: omit(
+    getSharedTableColumns(evaluations.resultType, {
+      [EvaluationResultableType.Boolean]: evaluationConfigurationBoolean,
+      [EvaluationResultableType.Number]: evaluationConfigurationNumerical,
+      [EvaluationResultableType.Text]: evaluationConfigurationText,
+    }),
+    ['id', 'createdAt', 'updatedAt'],
```
Comment (csansoon) [inline packages/core/src/repositories/evaluationsRepository.ts:37]: Here, the `metadata` attribute contains either the attributes from `evaluationMetadataLlmAsJudgeAdvanced` or `evaluationMetadataLlmAsJudgeSimple`, depending on the value of `evaluations.metadataType`. The `getSharedTableColumns` returns the correct SQL query to return the attributes from either one or the other depending on the actual value!

The same happens with `resultConfiguration`, which can be the attributes from `evaluationConfigurationBoolean`, `evaluationConfigurationNumerical` or `evaluationConfigurationText`, depending on the value of `evaluations.resultType`.

File: apps/web/src/actions/evaluations/create.ts [hunk 2]
```diff
@@ -34,13 +27,12 @@ export const createEvaluationAction = authProcedure
     { type: 'json' },
   )
   .handler(async ({ input, ctx }) => {
-    const result = await createEvaluation({
+    const result = await createAdvancedEvaluation({
```
Comment (geclos) [inline apps/web/src/actions/evaluations/create.ts:30]: what's an advanced evaluation

File: packages/core/drizzle/0091_bigint_evaluations_result_config.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE "latitude"."evaluation_configuration_numerical" ALTER COLUMN "min_value" SET DATA TYPE bigint;--> statement-breakpoint
+ALTER TABLE "latitude"."evaluation_configuration_numerical" ALTER COLUMN "max_value" SET DATA TYPE bigint;
```
Comment (geclos) [inline packages/core/drizzle/0091_bigint_evaluations_result_config.sql:2]: what types were they before?

File: packages/core/src/lib/compactObject.ts [hunk 4]
```diff
@@ -1,9 +1,7 @@
...
-    if (value !== undefined) newObj[key] = value
-  })
-
-  return newObj
+export const compactObject = (
+  obj: Record<string, unknown | undefined>,
+): Record<string, unknown> => {
+  return Object.fromEntries(
+    Object.entries(obj).filter(([_, value]) => value !== undefined),
+  )
```
Comment (geclos) [inline packages/core/src/lib/compactObject.ts:6]: is this changing behaviour? this is used in many places

File: packages/core/src/schema/models/evaluationConfigurationNumerical.ts [hunk 5]
```diff
@@ -7,8 +7,8 @@ export const evaluationConfigurationNumerical = latitudeSchema.table(
   'evaluation_configuration_numerical',
   {
     id: bigserial('id', { mode: 'number' }).notNull().primaryKey(),
-    minValue: numeric('min_value').notNull(),
-    maxValue: numeric('max_value').notNull(),
+    minValue: bigint('min_value', { mode: 'number' }).notNull(),
+    maxValue: bigint('max_value', { mode: 'number' }).notNull(),
```
Comment (geclos) [inline packages/core/src/schema/models/evaluationConfigurationNumerical.ts:11]: this type change does not cause data removal no?

File: apps/web/src/actions/evaluations/create.ts (hunk 2 above)
Comment (csansoon) [reply apps/web/src/actions/evaluations/create.ts:30]: An evaluation where the user defines the prompt (the ones we have now)

File: packages/core/src/services/evaluations/create.ts [hunk 6]
```diff
@@ -194,3 +225,68 @@ async function findProvider(workspace: Workspace, db: Database) {
...
+        prompt: promptWithProvider,
+        configuration,
+        templateId: metadata.templateId ?? null,
+      } as Omit<EvaluationMetadataLlmAsJudgeAdvanced, 'id'>,
+      resultType: configuration.type,
+      resultConfiguration,
+    },
+    db,
+  )
+}
```
Comment (geclos) [inline packages/core/src/services/evaluations/create.ts:292]: there's no way you are merging this without adding tests for this service 😅

File: packages/core/drizzle/0091_bigint_evaluations_result_config.sql (hunk 3 above)
Comment (csansoon) [reply packages/core/drizzle/0091_bigint_evaluations_result_config.sql:2]: "numeric", which for some reason is stored as a string. Anyways, this has to match with the value of `EvaluationResultableNumber`, which is a `bigint`

**Review by geclos — APPROVED** [review]
look good but add tests this is touching sensible code

File: packages/core/src/lib/compactObject.ts (hunk 4 above)
Comment (csansoon) [reply packages/core/src/lib/compactObject.ts:6]: Nope, same behaviour. Could just undo the change, just felt this was more elegant :)

File: packages/core/src/schema/models/evaluationConfigurationNumerical.ts (hunk 5 above)
Comment (csansoon) [reply packages/core/src/schema/models/evaluationConfigurationNumerical.ts:11]: No, there's still no rows whatsoever in this table. There will be from now on tho

File: packages/core/src/services/evaluations/create.ts (hunk 6 above)
Comment (csansoon) [reply packages/core/src/services/evaluations/create.ts:292]: There are tests for this service! They check that everything is still working as expected.

**Pull request merged**